presto-docs/src/main/sphinx/sql/select.rst

```diff
@@ -577,32 +578,99 @@ output expressions:
 
 Each expression may be composed of output columns or it may be an ordinal
 number selecting an output column by position (starting at one). The
-``ORDER BY`` clause is evaluated as the last step of a query after any
-``GROUP BY`` or ``HAVING`` clause. The default null ordering is ``NULLS LAST``,
-regardless of the ordering direction.
+``ORDER BY`` clause is evaluated after any ``GROUP BY`` or ``HAVING`` clause
+and before any ``OFFSET``, ``LIMIT`` or ``FETCH FIRST`` clause.
+The default null ordering is ``NULLS LAST``, regardless of the ordering direction.
 
-LIMIT Clause
-------------
+.. _offset-clause:
 
-The ``LIMIT`` clause restricts the number of rows in the result set.
-``LIMIT ALL`` is the same as omitting the ``LIMIT`` clause.
-The following example queries a large table, but the limit clause restricts
-the output to only have five rows (because the query lacks an ``ORDER BY``,
+OFFSET Clause
+-------------
+
+The ``OFFSET`` clause is used to discard a number of leading rows
+from the result set:
+
+.. code-block:: none
+
+    OFFSET count [ { ROW | ROWS } ]
+
+If the ``ORDER BY`` clause is present, the ``OFFSET`` clause is evaluated
+over a sorted result set, and the set remains sorted after the
+leading rows are discarded::
+
+    SELECT name FROM nation ORDER BY name OFFSET 22;
+
+.. code-block:: none
+
+          name
+    ----------------
+     UNITED KINGDOM
+     UNITED STATES
+     VIETNAM
+    (3 rows)
+
+Otherwise, it is arbitrary which rows are discarded.
+If the count specified in the ``OFFSET`` clause equals or exceeds the size
+of the result set, the final result is empty.
+
+LIMIT | FETCH FIRST Clause
+--------------------------
+
+The ``LIMIT`` and ``FETCH FIRST`` clauses restrict the number of rows
+in the result set.
+
+.. code-block:: none
+
+    LIMIT [ count | ALL ]
+
+.. code-block:: none
+
+    FETCH { FIRST | NEXT } [ count ] { ROW | ROWS } ONLY
+
+The following example queries a large table, but the ``LIMIT`` clause
+restricts the output to only have five rows (because the query lacks an ``ORDER BY``,
 exactly which rows are returned is arbitrary)::
 
     SELECT orderdate FROM orders LIMIT 5;
 
 .. code-block:: none
 
-     o_orderdate
-    -------------
-     1996-04-14
-     1992-01-15
-     1995-02-01
-     1995-11-12
-     1992-04-26
+     orderdate
+    ------------
+     1994-07-25
+     1993-11-12
+     1992-10-06
+     1994-01-04
+     1997-12-28
     (5 rows)
 
+``LIMIT ALL`` is the same as omitting the ``LIMIT`` clause.
+
+If the count is not specified in the ``FETCH FIRST`` clause, it defaults to 1.
```

Comment:
We can simplify this to
```
If the count is not specified in the ``FETCH FIRST`` clause, it defaults to ``1``::
```
(thus we don't need a second sentence to call out the example)